gobblin-modules/gobblin-http/src/main/java/gobblin/http/HttpRequestBuilder.java

```
@@ -0,0 +1,106 @@
+ *
+ * <p>
+ *   This basic implementation builds a write request from a single record. However, it has the extensibility to build
+ *   a write request from batched records, depending on specific implementation of {@link #buildWriteRequest(Queue)}
+ * </p>
+ */
+public class HttpRequestBuilder implements AsyncWriteRequestBuilder<GenericRecord, HttpUriRequest> {
+  private static final Logger LOG = LoggerFactory.getLogger(HttpRequestBuilder.class);
+  private final String urlTemplate;
+  private final String verb;
+
+  public HttpRequestBuilder(String urlTemplate, String verb) {
+    this.urlTemplate = urlTemplate;
+    this.verb = verb;
+  }
+
+  @Override
+  public AsyncWriteRequest<GenericRecord, HttpUriRequest> buildWriteRequest(Queue<BufferedRecord<GenericRecord>> buffer) {
+    return buildWriteRequest(buffer.poll());
+  }
+
+  /**
+   * Build a write request from a single record
+   */
+  private AsyncWriteRequest<GenericRecord, HttpUriRequest> buildWriteRequest(BufferedRecord<GenericRecord> record) {
+    if (record == null) {
+      return null;
+    }
+
+    AsyncWriteRequest<GenericRecord, HttpUriRequest> request = new AsyncWriteRequest<>();
+    HttpOperation httpOperation = HttpUtils.toHttpOperation(record.getRecord());
+
+    // Set uri
+    URI uri = HttpUtils.buildURI(urlTemplate, httpOperation.getKeys(), httpOperation.getQueryParams());
+    if (uri == null) {
+      return null;
+    }
+
+    RequestBuilder builder = RequestBuilder.create(verb.toUpperCase());
+    builder.setUri(uri);
+
+    // Set headers
+    Map<String, String> headers = httpOperation.getHeaders();
+    if (headers != null && headers.size() != 0) {
+      for (Map.Entry<String, String> header : headers.entrySet()) {
+        builder.setHeader(header.getKey(), header.getValue());
+      }
+    }
+
+    // Add payload
+    int bytesWritten = addPayload(builder, httpOperation.getBody());
+    if (bytesWritten == -1) {
+      return null;
```

Should you throw an error here?